src/main.rs

```diff
@@ -94,51 +155,25 @@ fn postgres_tunnel_cmd(postgres_config: &PostgresConfig, port: u16) -> Result<Op
         }
 }
 
-fn temp_file(extension: &str) -> PathBuf {
-    let mut dir = temp_dir();
-    let file_name = format!("{}.{}", Uuid::new_v4(), extension);
-
-    dir.push(file_name);
-
-    dir
-}
-
-fn postgres_pgbouncer_cmd(postgres_config: &PostgresConfig, port: u16, upstream_port: u16) -> Result<Command> {
-    let userlist_file_path_str = temp_file("txt");
-    let mut userlist_file = File::create(userlist_file_path_str.clone())?;
-    let ini_file_path_str = temp_file("ini");
-    let mut ini_file = File::create(ini_file_path_str.clone())?;
+fn postgres_pgbouncer_cmd(config: &PostgresConfig, port: u16, upstream_port: u16) -> Result<Command> {
+    let userlist_file_path_str = util::temp_file("txt");
+    let mut userlist_file = fs::File::create(&userlist_file_path_str)?;
+    let ini_file_path_str = util::temp_file("ini");
+    let mut ini_file = fs::File::create(&ini_file_path_str)?;
 
-    userlist_file.write_all(format!("\"{}\" \"{}\"", &postgres_config.user, &postgres_config.password).as_bytes())?;
+    userlist_file.write_all(format!("\"{}\" \"{}\"", &config.user, &config.password).as_bytes())?;
 
     // TODO change to md5 hash of password
     // TODO remove
-    println!("{}", ini_file_path_str.to_str().unwrap());
-    println!("\"{}\" \"{}\"", &postgres_config.user, &postgres_config.password);
-
-    let ini_content = format!(
-        r#"################## fig-cli pgbouncer configuration ##################
-
-[databases]
-{} = host=localhost port={} user={} dbname={} password={}
-
-[pgbouncer]
-listen_addr = 0.0.0.0
-listen_port = {}
-unix_socket_dir =
-auth_type = any
-pool_mode = transaction
-default_pool_size = 1
-ignore_startup_parameters = extra_float_digits
-
-################## end file ##################
-"#,
-        postgres_config.database,
-        upstream_port,
-        postgres_config.user,
-        postgres_config.database,
-        postgres_config.password,
-        port,
+    println!("{}", ini_file_path_str.display());
+    println!("\"{}\" \"{}\"", &config.user, &config.password);
+
+    let ini_content = format!(include_str!("../template/pgbouncer.toml.template"),
```

Comment:
The template files gets packaged in crates.io?

Reply:
@scirner22 They get compiled directly into the binary. The contents are interpolated at compile-time into a string. https://doc.rust-lang.org/std/macro.include_str.html

Reply:
What a great macro.